src/main/java/com/google/api/codegen/viewmodel/CallingForm.java

```diff
@@ -132,6 +133,42 @@ static RpcType fromMethodContext(MethodContext context) {
   private static final Table<TargetLanguage, RpcType, ImmutableList<CallingForm>>
       CALLING_FORM_TABLE =
           ImmutableTable.<TargetLanguage, RpcType, ImmutableList<CallingForm>>builder()
+              .put(
```

Comment:
What's the relationship between this table and the `CALLING_FORM_CLIENT_METHOD_TYPE_MAP` you have in the other files above? This one maps RPC types to CallingForms, the other one maps ClientMethodTypes to CallingForms. Why do we need both? Is this something we could simplify?

Reply:
`RpcType` is the proto/grpc layer and contains six values: `Unary`, `PagedStreaming`, `LongRunning`, `GrpcClientStreaming`, `GrpcServerStreaming`, `GrpcBidiStreaming`.

For Java and C#, one RPC method would have multiple overloads, and the type of the overload is determined by `ClientMethodType`. For example, a unary RPC would have at least three public method in Java client libraries: `RequestObjectMethod`, `FlattenedMethod, CallableMethod`. For all other languages there aren't multiple overloads, but because we generate all of them in one generator they are forced to be compatible with Java and C#.

`CallingForm` has some overlaps with `ClientMethodType` but not completely equivalent. For example, For Node.js LROs, there is just one client method type: `LongRunningOptionalArrayMethod`, but we have multiple calling forms because there are multiple patterns to call an LRO method in Node.js. But for Java and C#, `ClientMethodType` and `CallingForm` are _mostly_ one to one.

I see this as an unavoidable problem of having one generator work for seven languages - the terminologies have to be general, but they don't always make much sense when it comes to a specific language :(

Making it less bad, the enum `RpcType` is a private enum inside `CallingForm` and thus invisible outside, which might make it less confusing for maintainers of this generator :/